sorc/ncep_post.fd/MPI_FIRST.f

```diff
@@ -116,24 +142,69 @@ SUBROUTINE MPI_FIRST()
 !
 !     global loop ranges
 !
-      call para_range(1,jm,num_procs,me,jsta,jend)
+!      call para_range(1,jm,num_procs,me,jsta,jend)
+! GWVX temporary documentation
+!  para_range2 supports a 2D decomposition.  The rest of the post
+!  supports 1D still and the call here is the special case where each
+!  processor gets all of the longitudes in the latitude 1D subdomain
+!  jsta:jend.  The X decomposition will be specified by the third
+!  argument (currently 1) and the Y decoposition will be specified by
+!  the fourth argument (currently all of the ranks)   When X is
+!  subdivided the third and fourth arguments will have to be integral
+!  factors of num_procs and on 5/27/21 I am still working out a general
+!  way to do this if the user doesn't select the factors
+     ! call para_range2(im,jm,1,num_procs,me,ista,iend,jsta,jend)
+      call para_range2(im,jm,numx,num_procs/numx,me,ista,iend,jsta,jend)
```

Comment:
@GeorgeVandenberghe-NOAA I got failure of setting num_procs undividable by numx, e.g. num_procs=10, numx=3. Do you have a fix for this situation, or might add an exception handle for users?

Reply:
@BoCui-NOAA That would be great. Thanks!

Reply:
Wen, in code WRFPOST.f  I add a  reset of numx=1 if the remainder of num_procs/numx is not 0. The RT tests for numx=1, 4, 5, 7, 8 have been run and all passed successfully. @JesseMeng-NOAA @GeorgeVandenberghe-NOAA